packages/page-calendar/src/DayItem.tsx

```diff
@@ -46,62 +50,116 @@ function DayItem ({ className, item: { date, info, type } }: Props): React.React
+          s = 'Start of a new staking era {{id}}';
+          break;
 
         case 'stakingSlash':
-          return <><div className='itemDesc'>{t<string>('Application of slashes from era {{id}}', { replace: { id } })}</div>{typeLink}</>;
+          s = 'Application of slashes from era {{id}}';
+          break;
 
         case 'treasurySpend':
-          return <><div className='itemDesc'>{t<string>('Start of next spending period')}</div>{typeLink}</>;
+          s = 'Start of next spending period';
+          break;
 
         case 'societyChallenge':
-          return <><div className='itemDesc'>{t<string>('Start of next membership challenge period')}</div>{typeLink}</>;
+          s = 'Start of next membership challenge period';
+          break;
 
         case 'societyRotate':
-          return <><div className='itemDesc'>{t<string>('Acceptance of new members and bids')}</div>{typeLink}</>;
+          s = 'Acceptance of new members and bids';
+          break;
 
         default:
           return assertUnreachable(type);
       }
+
+      setDescription(id ? s.replaceAll('{{id}}', id) : s);
+
+      return id ? <><div className='itemDesc'>{t<string>(s, { replace: { id } })}</div>{typeLink}</> : (<><div className='itemDesc'>{s}</div>{typeLink}</>);
     },
     [info, t, type]
   );
 
+  function exportToCal (fileName: string, date: Date): void {
+    const startDate = dateCalendarFormat(date);
+    // For now just add 1 hour for each event
+    const endDate = dateCalendarFormat(new Date(new Date(date).setHours(new Date(date).getHours() + 1)));
+    const calData =
+      'BEGIN:VCALENDAR\n' +
+      'CALSCALE:GREGORIAN\n' +
+      'METHOD:PUBLISH\n' +
+      'PRODID:-//Test Cal//EN\n' +
+      'VERSION:2.0\n' +
+      'BEGIN:VEVENT\n' +
+      'UID:test-1\n' +
+      'DTSTART;VALUE=DATE:' + startDate + '\n' +
+      'DTEND;VALUE=DATE:' + endDate + '\n' +
+      'SUMMARY:' + description + '\n' +
+      'DESCRIPTION:' + description + '\n' +
+      'END:VEVENT\n' +
+      'END:VCALENDAR';
+    const fileNameIcs = encodeURI(fileName) + '.ics';
+    const data = new File([calData], fileNameIcs, { type: 'text/plain' });
+    const anchor = window.document.createElement('a');
+
+    anchor.href = window.URL.createObjectURL(data);
+    anchor.download = fileNameIcs;
+    document.body.appendChild(anchor);
+    anchor.click();
+    document.body.removeChild(anchor);
+    window.URL.revokeObjectURL(anchor.href);
+  }
+
+  function calendarIcon (date: Date): React.ReactNode {
+    if (date) {
+      return (<Button className='exportCal'
+        icon='calendar-plus'
+        onClick={() => exportToCal(description, date)} />);
+    } else {
+      return null;
+    }
+  }
+
   return (
     <div className={className}>
       <div className='itemTime'>{date.toLocaleTimeString().split(':').slice(0, 2).join(':')}</div>
       {desc}
+      {calendarIcon(date)}
```

Comment:
Sorry, this comment got lost (for some obscure reason). The above can just simply be (without extracting the function to file-level) -

```suggestion
      {date && (
        <Button
          className='exportCal'
          icon='calendar-plus'
          onClick={_exportCal}
      )}
```

(It assumes we have a `useCallback` with the name `_exportCal`)